Testing.Rmd

```diff
@@ -596,5 +1364,101 @@ There are a few things to bear in mind:
   32-bit versions of R) so use `expect_equal()` rather than
   `expect_identical()`.
 
-[tdd]:https://en.wikipedia.org/wiki/Test-driven_development
-[extreme-programming]:https://en.wikipedia.org/wiki/Extreme_programming
+CRAN revision thought dump:
+
+* Enrich description of platforms, i.e. say more than "Windows, Mac, Linux".
+  The check flavors: <https://cran.r-project.org/web/checks/check_flavors.html>.
+  - Draw a line to how this determines which "flavors" a package should be
+    checked on.
+    Depends on how broad the user base is and the "trickiness" of the package's
+    code (e.g. does it need compilation?).
+    Do a cost benefit analysis.
+    Connect that to which `usethis::use_github_action_check_*()` variant is
+    appropriate.
+* How to decide whether to let a test run on CRAN.
+  If the test can fail WITHOUT indicating a defect in your package that merits
+  removal from CRAN, do not let CRAN run the test. Examples:
+  - Accessing a website or API that could ever be down or have an
+    occasional error (so: any website or API on planet earth).
+  - Exact formatting of a message.
+  - File system work typical of an authentic user that touches files any where
```

Comment:
If you have to write to a file use `withr::local_temp_file()`. This will protect you against 99% of the most likely problems and if new problems arise we can fix centrally in one location.